GoogleForCreators/web-stories-wp
# Design System Storybook: Better buttons

## Summary

I initially built the buttons and then Sam and the design team updated them to have a square version and a smaller rectangle. Additionally, I didn't realize that sometimes we're going to have buttons on an unknown background (sounds obvious, but it wasn't when first looking) and so i needed to revisit how to get the focus styles to work. 

This PR catches all button related content for the new design system up to what figma currently has. 

![all the button varieties](https://user-images.githubusercontent.com/10720454/98878833-307e7480-2441-11eb-8185-3fc84c19577f.gif)
![lightmode pill](https://user-images.githubusercontent.com/10720454/98878842-34aa9200-2441-11eb-9d33-1ab1cf46c619.gif)
![darkmode pill](https://user-images.githubusercontent.com/10720454/98878843-35432880-2441-11eb-8ad1-c75829f0376e.gif)

## Relevant Technical Choices

- The button focus style has a 2px space between the button and the border/outline - this is really tricky because we need to allow a border radius to come through (so we can't use outline and outline-offset), gradients are out, box shadow is out too because we can't get a transparent space in there. Decided to use padding-box to fake the gap between the button and the border, this meant updating all of the sizing styles and a transparent border on all elements that use the focus style to avoid shifting content.  (thanks @embarks for helping with this idea!)

- The tab component turned into a pill - updated that here too (it's a type of button), i also got rid of some of the logic surrounding the 'tab' (deleted tabGroup completely). After seeing the updates Sam has made to the designs, pill without a ref makes wayyy more sense and if it's needed internal to an app we can add it there. I want to be agnostic here. 

## To-do

- I'll need to update the dialog PR for dashboard once this is merged to make sure button size small is getting passed for those instances. But that dialog pr can't get merged until colors are merged, and this PR is blocked by that. 

- Find the source of truth for the disabled background color: Note that the disabled button bg color is supposed to be `interactiveBg.disable` - this value in the color palette is `gray 80 #26292A`, the button component color shown is `#2F3131`. I went with the color palette as the source of truth for the time being since I know this was most recently updated.

## User-facing changes

Storybook changes only

## Testing Instructions

Open storybook, see designsSystem/components/button and designsStem/components/pill and compare to [figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3438%3A125786)

---



Fixes #
